tests/tools/fswalker/fswalker.go

```diff
@@ -0,0 +1,204 @@
+// +build linux
+
+// Package fswalker provides the checker.Comparer interface using FSWalker
+// walker and reporter.
+package fswalker
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"path/filepath"
+	"strings"
+
+	"github.com/golang/protobuf/proto"
+	"github.com/google/fswalker"
+	fspb "github.com/google/fswalker/proto/fswalker"
+
+	"github.com/kopia/kopia/tests/robustness/checker"
+	"github.com/kopia/kopia/tests/tools/fswalker/reporter"
+	"github.com/kopia/kopia/tests/tools/fswalker/walker"
+)
+
+var _ checker.Comparer = &WalkCompare{}
+
+// WalkCompare is a checker.Comparer that utilizes the fswalker
+// libraries to perform the data consistency check.
+type WalkCompare struct {
+	GlobalFilterMatchers []string
+}
+
+// NewWalkCompare instantiates a new WalkCompare and returns its pointer
+func NewWalkCompare() *WalkCompare {
+	return &WalkCompare{
+		GlobalFilterMatchers: []string{
+			"ctime:",
+			"atime:",
+			"mtime:",
+		},
+	}
+}
+
+// Gather meets the checker.Comparer interface. It performs a fswalker Walk
+// and returns the resulting Walk as a protobuf Marshaled buffer.
+func (chk *WalkCompare) Gather(ctx context.Context, path string) ([]byte, error) {
+	walkData, err := walker.WalkPathHash(ctx, path)
+	if err != nil {
+		return nil, err
+	}
+
+	err = rerootWalkDataPaths(walkData, path)
+	if err != nil {
+		return nil, err
+	}
+
+	// Store the walk data along with the snapshot ID
+	b, err := proto.Marshal(walkData)
+	if err != nil {
+		return nil, err
+	}
+
+	return b, nil
+}
+
+// Compare meets the checker.Comparer interface. It performs a fswalker Walk
+// on the provided file path, unmarshals the comparison data as a fswalker Walk,
+// and generates a fswalker report comparing the two Walks. If there are any differences
+// an error is returned, and the full report will be written to the provided writer
+// as JSON.
+func (chk *WalkCompare) Compare(ctx context.Context, path string, data []byte, reportOut io.Writer) error {
+	beforeWalk := &fspb.Walk{}
+
+	err := proto.Unmarshal(data, beforeWalk)
+	if err != nil {
+		return err
```

Comment:
can we use use errors.Wrap() instead of bare error propagation, same thing below?